x/src/cargo.rs

```diff
@@ -0,0 +1,94 @@
+use crate::Result;
+use anyhow::anyhow;
+use std::{
+    ffi::{OsStr, OsString},
+    path::Path,
+    process::{Command, Output, Stdio},
+};
+
+pub struct Cargo {
+    inner: Command,
+    pass_through_args: Vec<OsString>,
+}
+
+impl Cargo {
+    pub fn new<S: AsRef<OsStr>>(command: S) -> Self {
+        let mut inner = Command::new("cargo");
+        inner.arg(command);
+        Self {
+            inner,
+            pass_through_args: Vec::new(),
+        }
+    }
+
+    pub fn all(&mut self) -> &mut Self {
+        self.inner.arg("--all");
+        self
+    }
+
+    pub fn all_features(&mut self) -> &mut Self {
+        self.inner.arg("--all-features");
+        self
+    }
+
+    pub fn current_dir<P: AsRef<Path>>(&mut self, dir: P) -> &mut Self {
+        self.inner.current_dir(dir);
+        self
+    }
+
+    pub fn packages<I, S>(&mut self, packages: I) -> &mut Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<OsStr>,
+    {
+        for p in packages {
+            self.inner.arg("-p");
+            self.inner.arg(p);
+        }
+        self
+    }
+
+    pub fn exclusions<I, S>(&mut self, exclusions: I) -> &mut Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<OsStr>,
+    {
+        for e in exclusions {
+            self.inner.arg("--exclude");
+            self.inner.arg(e);
+        }
+        self
+    }
+
+    pub fn pass_through<I, S>(&mut self, args: I) -> &mut Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<OsStr>,
+    {
+        for a in args {
+            self.pass_through_args.push(a.as_ref().to_owned());
```

Comment:
Not exactly there's a number of reasons, 1) you still need to call `as_ref()` since extend or `cloned` would handle that 2) `OsStr` doesn't impl clone so `cloned()` doesn't work.